sycl/include/CL/sycl/accessor.hpp

```diff
@@ -382,24 +392,29 @@ class image_accessor
   };
 
   //  Available only when: (accessTarget == access::target::image ||
-  //  accessTarget == access::target::host_image) && accessMode ==
-  //  access::mode::read
+  //  accessTarget == access::target::host_image) && (accessMode ==
+  //  access::mode::read || accessMode == access::mode::read_write)
+  //  The appropriate allowed AccessMode for the AccessTarget is checked in the
+  //  static_assert in the class.
   template <typename CoordT, int Dims = Dimensions,
             typename = detail::enable_if_t<
                 (Dims > 0) && (IsValidCoordDataT<Dims, CoordT>::value) &&
-                (IsImageAcc || IsHostImageAcc) && IsImageAccessRead>>
+                (IsImageAcc || IsHostImageAcc) && IsImageAccessAnyRead>>
```

Comment:
This allows `read_write` access on the device, which is not supposed to be allowed.

Reply:
added the requested changes.